addon/-private/properties/descriptor.js

```diff
@@ -0,0 +1,55 @@
+import { throwBetterError } from '../better-errors';
+
+const NOT_A_FUNCTION_ERROR = 'Argument passed to `descriptor` must be a function.';
+
+/**
+ * Creates a descriptor whose value is determined by the passed-in function.
+ * The passed-in function must not be bound and must not be an arrow function,
+ * as this will prevent the function from being called with the correct context.
```

Comment:
In the `test` documentation (#318), I described the conditions in which an arrow function can be used successfully, but I can't think of a reason why an arrow function would ever be appropriate with `descriptor`...I can't imagine writing a `descriptor` that doesn't access page object properties via `this`. Thoughts?